File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int[] res = new int[2];
        int anchor = 0 ;
        HashMap<Integer,Integer> m = new HashMap<>();
        for(int i=0; i<n; i++){
            anchor = target - nums[i];
            if(m.containsKey(anchor)==true){
                res[0]= m.get(anchor);
                res[1]=i;
            }else{
                m.put(nums[i],i);
            }
        }        
        return res;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res, cur = [], []

        def dfs(i, cur_target):
            if cur_target == 0:
                res.append(cur.copy())
                return
            elif cur_target < 0 or i == len(nums):
                return
            cur.append(nums[i])
            cur_target -= nums[i]
            dfs(i, cur_target)
            cur.pop()
            cur_target += nums[i]
            dfs(i+1, cur_target)
        
        dfs(0, target)
        return res

